waku/v2/protocol/rln/group_manager/dynamic/web3.go

```diff
@@ -112,7 +112,17 @@ func (gm *DynamicGroupManager) HandleGroupUpdates(ctx context.Context, handler R
 }
 
 func (gm *DynamicGroupManager) loadOldEvents(ctx context.Context, rlnContract *contracts.RLN, handler RegistrationEventHandler) error {
-	events, err := gm.getEvents(ctx, 0, nil)
+
```

Comment:
It should be possible to retrieve events with a full node because these type of nodes contain all transaction receipts/logs. A full node can still be costly so I agree that we'll need to come up with a solution for obtaining the merkle tree changes, instead of relying only on the event logs (although for the time being, it should be fine with using infura/pocket)